File: Python/LinkedList/LinkedList.py
class Node:
    def __init__(self, data):
        self.data= data
        self.next = None;

class LinkedList:
    def __init__(self):
        self.head= None;
        self.tail = None;
    def add(self, data):
        newNode = Node(data)
        if self.head == None:
            self.head = newNode
            self.tail = newNode
        else:
            self.tail.next = newNode
            self.tail = newNode

    def showNode(self):
        current = self.head
        while current != None:
            print(f"Data: {current.data}")
            current = current.next
    def search(self, data):
        current = self.head
        while current != None:
            if current.data == data:
                return True
            current = current.next
        return False
    
    def delete(self,data):
        if self.head == None:
            return
        if self.head.data == data:
            self.head = self.head.next
            if self.head == None:
                self.tail  = None
        return None
    
    def size(self):
        current = self.head
        count = 0
        while current != None:
            count += 1
            current = current.next
        return count



linkList = LinkedList()
linkList.add(1)
linkList.add(2)
linkList.add(3)
print(linkList.delete(2))
linkList.showNode()

File: Java/LinkedList/LinkedList.java
package Java.LinkedList;

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class LinkedList {
    private Node head;
    private Node tail;

    public LinkedList() {
        this.head = null;
        this.tail = null;
    }

    public void add(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            tail = newNode;
        } else {
            tail.next = newNode;
            tail = newNode;
        }
    }

    public void showNode() {
        Node current = head;
        while (current != null) {
            System.out.println("The data is " + current.data);
            current = current.next;
        }
    }

    public boolean search(int data) {
        Node current = head;
        while (current != null) {
            if (current.data == data) {
                return true;
            }
            current = current.next;

        }
        return false;
    }

    public void delete(int data) {
        if (head == null) {
            return;
        }
        if (head.data == data) {
            head = head.next;
            if (head == null) {
                tail = null;
            }
            return;
        }
    }

    public int size() {
        Node current = head;
        int count = 0;
        while (current != null) {
            count++;
            current = current.next;

        }
        return count;
    }

    public static void main(String[] args) {

    }
}
